TuneUp/TuneUpWindowViewModel.cs

```diff
@@ -127,6 +157,8 @@ public TuneUpWindowViewModel(ViewLoadedParams p)
                 CurrentWorkspace = p.CurrentWorkspaceModel as HomeWorkspaceModel;
                 ResetProfiledNodes();
             }
+            // Saving UI context so later when we touch the collection, it is still performed in the same context
+            uiContext = SynchronizationContext.Current;
```

Comment:
I think alternatively, you can just use the application dispatcher, or the dispatcher from the DynamoWindow ... it will all be the same if I'm not crazy.